Lib/unittest/test/test_case.py

```diff
@@ -1273,6 +1273,19 @@ def testAssertRaisesNoExceptionType(self):
         with self.assertRaises(TypeError):
             self.assertRaises((ValueError, object))
 
+    def testAssertRaisesRefcount(self):
```

Comment:
I don't see any reference cycle when using the context manager protocol.

It seems like bpo-23890 was very specific to the function API: assertRaises(ValueError, func) since you pass func to assertRaises().

```
with self.assertRaises(ValueError):
    func()
```

Here `func` variable may also leak in the traceback objects linked to the raisen exception. But again, I'm unable to see any leak here without my fix.

If you still consider that an unit test is needed, please comment http://bugs.python.org/issue23890